packages/dds/tree/src/test/simple-tree/api/treeNodeApi.spec.ts

```diff
@@ -3544,6 +3544,138 @@ describe("treeNodeApi", () => {
 			}
 		});
 	});
+
+	describe("ensureType", () => {
+		const sf = new SchemaFactory("test");
+		class Son extends sf.object("Son", { value: sf.number }) {}
+		class Daughter extends sf.object("Daughter", { value: sf.number }) {}
+		class Parent extends sf.object("Parent", {
+			child: sf.optional([Son, Daughter]),
+		}) {}
+		it("returns the same value that was passed in", () => {
+			const child = { value: 3 };
+			const son = TreeAlpha.ensureSchema(Son, child);
+			assert.equal(son, child);
+		});
+
+		it("allows leaf types", () => {
```

Comment:
Should add a fluid handle test case too.

Reply:
Do you know how to make mock handles? I don't :D

Reply:
There's a helper class `MockHandle` in `@fluidframework/test-runtime-utils` that you can use. We've used that elsewhere in tree, I think.

Reply:
Thanks, done.